aiohttp/web_fileresponse.py

```diff
@@ -122,29 +132,31 @@ async def _precondition_failed(
         return await super().prepare(request)
 
     def _get_file_path_stat_and_gzip(
-        self, check_for_gzipped_file: bool
-    ) -> Tuple[pathlib.Path, os.stat_result, bool]:
-        """Return the file path, stat result, and gzip status.
+        self, accept_encoding: str
+    ) -> Tuple[pathlib.Path, os.stat_result, Optional[str]]:
+        """Return the file path, stat result, and possible compression type.
 
         This method should be called from a thread executor
         since it calls os.stat which may block.
         """
```

Comment:
Let's make the title shorter and not imply that `None` is an impossible type. Instead, clarify that `None` is a sentinel, semantically referring to an uncompressed file:
```suggestion
        """Return the file path, stat result, and compression type.

        If no compressed copy of the file is detected, in that case,
        the compression is represented as :py:data:`None`.

        This method should be called from a thread executor
        since it calls os.stat which may block.
        """
```

Reply:
Took your suggestion, but changed the wording as yours is not technically correct.

Reply:
I don't see any technical difference. You're referring to an implementation detail (something happens if something is returned, internally) and I'm trying to assign a high-level to a sentinel constant. But alright, it's better than before.

Reply:
Ah, I see. You're also replacing compression with encoding. That's fair.

Reply:
You wrote:

> If no compressed copy of the file is detected, ...

Whether or not a compressed file exists is not the only condition; its encoding also has to be supported and acceptable by the client.